build.rs

```diff
@@ -0,0 +1,55 @@
+use std::env;
+use std::fs;
+use std::fs::File;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::path::Path;
+use std::process;
+
+fn main() {
+    static ENV_VAR: &str = "PLATFORM";
+    static FILE_NAME: &str = "platform";
+
+    let platform_name = read_board_name_from_env_var(ENV_VAR)
+        .or_else(|| read_board_name_from_file(FILE_NAME))
+        .unwrap_or_else(|| {
+            println!(
+                "No platform specified. Either set the environment \
+                 variable {} or create a file named `{}`",
+                ENV_VAR, FILE_NAME
+            );
+            process::exit(1);
+        });
```

Comment:
Could we print all of the supported platforms at this point? That would help people figure out what they are doing wrong.

Reply:
I updated the behavior slightly
- If you do not specify a platform, you will get a warning.
- If you specify a "non-supported" platform, it will yell at you that the corresponding linker file is not available. It should be pretty clear that by listing all linker files you get a feeling about which platform is supported.